sapi/cli/php_cli_server.c

```diff
@@ -1791,9 +1839,13 @@ static int php_cli_server_client_ctor(php_cli_server_client *client, php_cli_ser
 	}
 	php_http_parser_init(&client->parser, PHP_HTTP_REQUEST);
 	client->request_read = 0;
+
+	client->last_header_element = NONE;
 	client->current_header_name = NULL;
 	client->current_header_name_len = 0;
-	client->current_header_name_allocated = 0;
```

Comment:
This initialization should not be removed. Alternatively or additionally the variable should be initialized in https://github.com/php/php-src/pull/2736/files#diff-d705e80d519f022dee38ccc6c177a926R1636.